<?xml version="1.0"?>
<launch>

    <!-- start map_server -->
    <node name="map_server" pkg="map_server" type="map_server" args="/home/niranjan/catkin_ws/src/inria_hbrs/inria_hbrs/environments/maps/scenario_1.yaml">  
  	  <param name="frame_id" type="str" value="/map"  />
    </node>


    <!-- start amcl -->
    <node pkg="amcl" type="amcl" name="amcl">
	  <!--<remap from="scan" to="scan_unified" />-->
	  <!--<remap from="scan" to="p3dx/laser/scan"/>-->
	  <param name="odom_model_type" value="omni"/>
		
	  <param name="gui_publish_rate" value="-1.0"/>
	  <param name="laser_max_beams" value="30"/>
	  <param name="laser_max_range" value="29.5"/>
	  <param name="min_particles" value="100"/>
	  <param name="max_particles" value="5000"/>
	  <param name="update_min_d" value="0.2"/>
	  <!--param name="update_min_a" value="1.0"/-->
	  <param name="kld_err" value="0.01"/>
	  <param name="kld_z" value="0.99"/>
	  <param name="odom_alpha1" value="0.2"/>
	  <param name="odom_alpha2" value="0.2"/>
	  <!-- translation std dev, m -->
	  <param name="odom_alpha3" value="0.2"/>
	  <param name="odom_alpha4" value="0.2"/>
	  <param name="odom_alpha5" value="0.2"/>
	  <param name="laser_z_hit" value="0.95"/>
	  <param name="laser_z_short" value="0.1"/>
	  <param name="laser_z_max" value="0.05"/>
	  <param name="laser_z_rand" value="0.05"/>
	  <param name="laser_sigma_hit" value="0.2"/>
	  <param name="laser_lambda_short" value="0.1"/>
	  <param name="laser_model_type" value="likelihood_field"/>
	  <!-- <param name="laser_model_type" value="beam"/> -->
	  <param name="laser_likelihood_max_dist" value="2.0"/>
	  <param name="update_min_d" value="0.2"/>
	  <param name="update_min_a" value="0.5"/>
	  <param name="odom_frame_id" value="/odom"/>
	  <param name="base_frame_id" value="/base_link"/>
	  <param name="global_frame_id" value="/map"/>
	  <param name="resample_interval" value="2"/>
	  <param name="transform_tolerance" value="0.1"/>
	  <param name="recovery_alpha_slow" value="0.0"/>
	  <param name="recovery_alpha_fast" value="0.0"/>
    </node>


    <!-- Launch laser_filter node -->
    <node pkg="laser_filters" type="scan_to_scan_filter_chain" output="screen" name="laser_filter">	       
	  <rosparam command="load" file="$(find pioneer3dx_navigation)/config/laser_filter.yaml" />
    </node>


    <!-- Launch move_base -->
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen" clear_params="true" >
          <rosparam file="$(find pioneer3dx_navigation)/config/costmap_common_params.yaml" command="load" ns="global_costmap" />
          <rosparam file="$(find pioneer3dx_navigation)/config/costmap_common_params.yaml" command="load" ns="local_costmap" />
    	  <rosparam file="$(find pioneer3dx_navigation)/config/local_costmap_params.yaml" command="load" />
    	  <rosparam file="$(find pioneer3dx_navigation)/config/global_costmap_params.yaml" command="load" />
          <rosparam file="$(find pioneer3dx_navigation)/config/base_local_planner_params.yaml" command="load" />  
          <param name="planner_frequency" value="10.0"/>
	  
          <remap from="/move_base/global_costmap/costmap" to="/global_costmap/costmap" />
          <remap from="/move_base/global_costmap/costmap_updates" to="/global_costmap/costmap_updates" />
    </node>


    <!-- Launch global_planner -->
    <node pkg="semantic_planner_global" type="semantic_planner_global" respawn="false" name="semantic_planner_global" output="screen" clear_params="true">
          <rosparam file="$(find semantic_planner_global)/cfg/costmap_common_params.yaml" command="load" ns="global_costmap" />
	  <rosparam file="$(find semantic_planner_global)/cfg/global_costmap_params.yaml" command="load" />
    </node>


    <!-- Launch local_planner -->
    <node pkg="semantic_planner_local" type="semantic_planner_local" respawn="false" name="semantic_planner_local" output="screen" clear_params="true">
          <rosparam file="$(find semantic_planner_local)/cfg/costmap_common_params.yaml" command="load" ns="local_costmap" />
	  <rosparam file="$(find semantic_planner_local)/cfg/local_costmap_params.yaml" command="load" />
	  <rosparam file="$(find semantic_planner_local)/cfg/semantic_planner_local_params.yaml" command="load"/>
    </node>

    <!-- Launch semantic_localization -->
    <node pkg="semantic_localization" type="semantic_localization_node" respawn="false" name="semantic_localization" output="screen" clear_params="true">

    </node>

    <!-- Launch goal monitor -->
    <node pkg="goal_monitor" type="goal_monitor_node" respawn="false" name="goal_monitor" output="screen" clear_params="true">
       
    </node>


</launch>
